forch/endpoint_handler.py

```diff
@@ -0,0 +1,93 @@
+"""Simple endpoint handler for remote OVS switching fabrics."""
+
+from concurrent import futures
+import logging
+import subprocess
+import time
+import grpc
+
+from forch.utils import get_logger
+
+try:
+    import daq.proto.session_server_pb2_grpc as server_grpc
+    from daq.proto.session_server_pb2 import SessionResult
+    from daq.proto.session_server_pb2_grpc.server_grpc import SessionServerServicer
+except ImportError as e:
+    class SessionServerServicer:
+        """Dummy class for weak import"""
+
+DEFAULT_SERVER_PORT = 50111
+DEFAULT_BIND_ADDRESS = '0.0.0.0'
+DEFAULT_VXLAN_PORT = 4789
+DEFAULT_VXLAN_VNI = 0
+
+VXLAN_CONFIG_CMD = 'sudo ovs-vsctl set interface vxlan type=vxlan '
+VXLAN_CONFIG_OPTS = 'options:remote_ip=%s options:key=%s options:dst_port=%s'
+
+class EndpointHandler:
+    """Class to handle endpoint updates"""
+
+    def __init__(self, target_ip):
+        self._logger = get_logger('endpproxy')
+        server_port = DEFAULT_SERVER_PORT
+        address = f'{target_ip}:{server_port}'
+        self._logger.info('Proxy requests to %s', address)
+        channel = grpc.insecure_channel(address)
+        self._stub = server_grpc.SessionServerStub(channel)
+
+    def process_endpoint(self, endpoint):
+        """Handle an endpoint request"""
+        self._logger.info('Process request for %s', endpoint.ip)
+        self._stub.ConfigureEndpoint(endpoint)
```

Comment:
I am confused about this setup. Is this for setuping up a vxlan tunnel to DAQ? then why is there the EndpointServer here to run a command... 
Also ConfigureEndpoint was previously defined in DAQ's grpc; it has since been removed so there is some disconnect somewhere.

Reply:
Right, I remember now. This is needed for some for some of the testing setups. I added it back into DAQ and a (lengthy) comment at the top of this file explaining what it's used for.